test/Microsoft.Azure.WebJobs.Host.EndToEndTests/SingletonEndToEndTests.cs

```diff
@@ -809,80 +816,77 @@ public override string Resolve(string name)
 
         private class TestFixture : IDisposable
         {
-            private Lazy<StorageAccount> _storageAccountLazy = new Lazy<StorageAccount>(GetStorageAccount);
+            public BlobServiceClient BlobServiceClient;
+            public BlobServiceClient SecondaryBlobServiceClient;
+            public QueueServiceClient QueueServiceClient;
 
-            public StorageAccount StorageAccount => _storageAccountLazy.Value;
 
             public TestFixture()
-            {
-                CloudBlobClient blobClient = StorageAccount.CreateCloudBlobClient();
-                _lockDirectory = blobClient.GetContainerReference("azure-webjobs-hosts").GetDirectoryReference("locks");
-
-                string secondaryConnectionString = GetConfigurationString($"AzureWebJobs{Secondary}");
-                var secondaryStorageAccount = CloudStorageAccount.Parse(secondaryConnectionString);
-                blobClient = secondaryStorageAccount.CreateCloudBlobClient();
-                _secondaryLockDirectory = blobClient.GetContainerReference("azure-webjobs-hosts").GetDirectoryReference("locks");
-            }
-
-            private static StorageAccount GetStorageAccount()
             {
                 // Create a default host since we know that's where the account
                 // is coming from
                 IHost host = new HostBuilder()
                     .ConfigureDefaultTestHost(b =>
                     {
-                        b.AddAzureStorage();
+                        b.AddAzureStorageCoreServices();
+
+                        // Necessary to manipulate Blobs/Queues for these tests
```

Comment:
Why is this necessary? Shouldn't be. Also the previous code was calling AddAzureStorage to add queues/blobs extensions. Don't you need to call the corresponding Track 2 methods here to add the extensions?

Reply:
The Track2 methods aren't needed because this isn't using any Queue/Blob bindings/triggers. We add the BlobServiceClientProvider/QueueServiceClientProvider as convenience methods to create blobs/queues for the tests, as opposed to just using the BlobContainerClient/QueueClient constructors directly. It gives us some consistency in how we treat configuration and connection string sourcing.